geode-core/src/main/java/org/apache/geode/management/internal/cli/GfshParser.java

```diff
@@ -203,6 +203,8 @@ public GfshParseResult parse(String userInput) {
 
     // this tells the simpleParser not to interpret backslash as escaping character
     rawInput = rawInput.replace("\\", "\\\\");
+    // this allows the usage of home shorthand across platforms
+    rawInput = rawInput.replace("~", System.getProperty("user.home"));
```

Comment:
I don't think you intend for `/~/`, `f~oo`, or other combinations of '~' to be replaced. The '~' should only be replaced if it anchors the beginning of the path and is immediately followed by a path separator. Something like this regex `^~(?:\/)` (untested) should do the trick.